prototype2/common/Producer.h

```diff
@@ -10,14 +10,24 @@
 #pragma once
 
 #include <librdkafka/rdkafkacpp.h>
+#include <common/Buffer.h>
+#include <functional>
```

Comment:
needed in h file or move to cpp?

Reply:
We do need them here. Buffer is a template class, so it must be defined. Functional is for the alias on line 83